spec/did-method-spec-template.html

```diff
@@ -227,10 +173,152 @@ <h3>Read (Resolve)</h3>
+
         <pre>
+          {
+            ...
+            'data': '{"endpoint":{"endpoint":"https://example.com/endpoint"}}',
+            'dest': 'AH4RRiPR78DUrCWatnCW2w' &lt; The DID being read/resolved&gt;, 
+            'raw': 'endpoint',
+            ...
+          }
         </pre>
 
+        <p>The value of the inner "endpoint" item ("https://example.com/endpoint" in the example above) is the service endpoint for the DID.</p>
+
+        <p>Historically on the Sovrin MainNet (through June 2021), the format of the `endpoint` ATTRIB has been simply a single name ("endpoint") value (a URL) pair, as shown above.    
+        An extended version of the `endpoint` ATTRIB can be written to an Indy ledger by a DID Controller to provide additional control over the generation of the DID Document.
+        The format is as follows:</p>
+
+        ```JSON
+        <pre>
+        {
+          "endpoint": "https://example.com",
+          "types": [ "endpoint", "did-communication", "DIDComm" ],
+          "routingKeys": [ "did:key12345", "did:key12345" ]
+        }
+        </pre>
+
+        <p>This corresponds to the raw value in the "ATTRIB" transaction as:</p>
+
+        <pre>
+        "raw": "{\"endpoint\":{\"endpoint\":\"https://example.com\",\"types\":[\"endpoint\",\"did-communication\",\"DIDComm\" ],\"routingKeys\":[\"did:key12345\",\"did:key12345\"]}}"
+        </pre>
+        
+        <p>A `did:sov` resolver MUST process the `endpoint` ATTRIB (extended or not) as follows:</p>
+
+        <ul>
+        <li>If the `types` field is present it's entries MUST be `endpoint`, `did-communication` and/or `DIDComm`, generating entries in the DID Document `service` block.</li>
+        <ul><li>The three service block formats corresponding to the three types is defined in the next section of this document.</li></ul>
+        <li>If the endpoint is defined and the `types` item is NOT present, empty or contains any other values, default the `types` to `"routingKeys": [ "endpoint", "did-communication" ]`.</li>
+        <ul><li>Note that any DID wanting to include a `DIDComm` service entry MUST use the extended version of the "endpoint" ATTRIB with the `types` item included.</li></ul>
+        <li>If the `routingKeys` item is present, use its value as the array for `routingKeys` in the `did-communication` and `DIDComm` type service blocks.</li>
+        <li>If the `routingKeys` entry is not present, default to the it being an empty array (e.g. `"routingKeys": [ ]`)</li>
+        </ul>
+
+        <h4>Resolver DID Document Format</h4>
+
+        <p>The DID Document is generated by a resolver (not by the Indy/Sovrin ledger) by taking the GET_NYM and optional GET_ATTRIB/endpoint data and 
+          rendering a DID Document as follows. See the notes following the rendering.
+        </p>
+
+        <pre>
+          {
+            "@context": [
+              "https://www.w3.org/ns/did/v1",
+              "https://didcomm.org/messaging/contexts/v2"
+            ],
```

Comment:
```suggestion
            "@context": [
              "https://www.w3.org/ns/did/v1",
              "https://didcomm.org/messaging/contexts/v2",
              "https://w3id.org/security/suites/ed25519-2020/v1",
              "https://w3id.org/security/suites/x25519-2020/v1"
            ],
```

This is the new approach -> Use more, but smaller, more focused JSON-LD contexts, instead of a single huge JSON-LD context that defines everything.